plugins/schedule/base/lib/model/ScheduleEvent.php

```diff
@@ -240,12 +238,18 @@ public function updateStartEndTimeOfFollowerEvents()
 				KalturaLog::err("Event $linkedByEventId not found");
 				continue;
 			}
-			$linkedEvent->setStartDate(strtotime($this->getEndDate()) + $linkedEvent->getLinkedTo()->getOffset());
-			$linkedEvent->setEndDate(strtotime($linkedEvent->getStartDate()) + $linkedEvent->getDuration());
+			$linkedEvent->shiftEvent(strtotime($this->getEndDate()), $linkedEvent->getLinkedTo()->getOffset(), $linkedEvent->getDuration());

```

Comment:
no need to send duration, since it is a core object property take from the object itself